CMakeLists.txt

```diff
@@ -2,15 +2,47 @@ cmake_minimum_required(VERSION 2.8.6)
 project(toxcore)
 include(CTest)
 
+set(CMAKE_MODULE_PATH ${toxcore_SOURCE_DIR}/cmake)
+
+################################################################################
+#
+# :: Version management
+#
+################################################################################
+
 # This version is for the entire project. All libraries (core, av, ...) move in
 # versions in a synchronised way.
 set(PROJECT_VERSION_MAJOR "0")
 set(PROJECT_VERSION_MINOR "1")
 set(PROJECT_VERSION_PATCH "4")
-set(PROJECT_VERSION
-  "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+
+# Update versions in various places
+# This must be run before setting SOVERSION, because SOVERSION is read from
+# a file generated by this script
+find_program(SHELL NAMES sh dash bash zsh)
+if(SHELL)
+  execute_process(
+      COMMAND ${SHELL} ${toxcore_SOURCE_DIR}/other/version-sync
```

Comment:
@iphydf can you please clarify why the version-sync script is run by cmake and not as a command on its own?